api/src/controllers/collection-controller.js

```diff
@@ -283,57 +282,91 @@ exports.unpublishCollections = async function (mineId, auth_payload) {
 
         const records = await nrpti.aggregate(recordAggregate).toArray();
 
-        // Unpublish all documents of all records.
+        let error = null;
+        // runs each function, last arg is callback, others are results from previous step
+        // any error in a step should cause the rest of the steps to be skipped and log that error
         for (const record of records) {
+          defaultLog.debug('publishing record docs, recordId: ', JSON.stringify(record._id) );
           if (record.documents && record.documents.length) {
-            promises.push(nrpti.updateMany({ _id: { $in: record.documents.map(doc => doc._id) }, write: { $in: auth_payload.realm_access.roles } }, { $pull: { read: 'public' } }));
 
-            // Update the S3 object properties for each document.
             for (const document of record.documents) {
               if (document.key) {
-                promises.push(unpublishS3Document(document.key));
+                try {
+                  await unpublishS3Document(document.key);
+                } catch (err) {
+                  error = `failed to unpublish record document in S3, recordId: ${record.recordName}`
+                  // log s3 sdk error
+                  defaultLog.error(err.message)
+                  throw new Error(error)
+                }
               }
             }
+
+            defaultLog.debug('unpulishing record doc objs, recordId: ', JSON.stringify(record._id))
+            try {
+              await nrpti.updateMany({ _id: { $in: record.documents.map(doc => doc._id) }, write: { $in: auth_payload.realm_access.roles } }, { $pull: { read: 'public' } })
+            } catch (err) {
+              error = `Could not unpublishing record document obj: ${record.recordName},  err: ${err}`;
```

Comment:
should be "unpublish" instead of "unpublishing" , same for other instances of this